Compiler error when appending to the z axis of a width and height fixed 3d array

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 b79b4bf.b383cce
OS: windows, Microsoft Windows 11 Pro v26100 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: E:\Projects\advent2024\day14
vexe: C:\Users\drake\AppData\Local\Programs\V\V04\v.exe
vexe mtime: 2024-12-03 02:00:11

vroot: OK, value: C:\Users\drake\AppData\Local\Programs\V\V04
VMODULES: OK, value: C:\Users\drake\.vmodules
VTMP: OK, value: C:\Users\drake\AppData\Local\Temp\v_0

Git version: git version 2.41.0.windows.3
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


emcc version: Error: 'emcc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc: N/A
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg bug.v`

```v
fn main() {
	// create a 2x2x* 3d array that holds int values
	mut space := [2][2][]int{}
	// append 0 to the z axis of the cell at 1,1
	space[1][1] << 0
	// print the value at 1,1
	print(space[1][1])
}
```

**What did you expect to see?**

[0]

**What did you see instead?**

```
================== C compilation error (from tcc): ==============
cc: C:/Users/drake/AppData/Local/Temp/v_0/bug.01JF1YDJYMTDVTG6Y51D71DDF6.tmp.c:7190: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/drake/AppData/Local/Temp/v_0/bug.01JF1YDJYMTDVTG6Y51D71DDF6.tmp.c:13896: error: too few arguments to function
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzVkMjhlOTFlMDYzMmRkMDhlMzkwZDgiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.MmXu8pxhrqg-gX5ajTbc9LwJLhb4dh9N5ymU0d96xCM">Huly&reg;: <b>V_0.6-21593</b></a></sub>

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -348,13 +348,16 @@ fn (mut g Gen) index_of_fixed_array(node ast.IndexExpr, sym ast.TypeSymbol) {
 		g.writeln(';')
 		g.past_tmp_var_done(past)
 	} else if node.left is ast.IndexExpr && node.left.is_setter {
-		past := g.past_tmp_var_new()
+		line := g.go_before_last_stmt().trim_space()
+		g.empty_line = true
+		tmp_var := g.new_tmp_var()
 		styp := g.styp(node.left_type)
-		g.write('${styp}* ${past.tmp_var} = &')
+		g.write('${styp}* ${tmp_var} = &')
 		g.expr(node.left)
 		g.writeln(';')
+		g.write(line)
 		g.write('(*')
-		g.past_tmp_var_done(past)
+		g.write(tmp_var)
 		g.write(')')
 	} else {
 		if is_fn_index_call {

--- a/vlib/v/tests/left_shift_array_fixed_test.v
+++ b/vlib/v/tests/left_shift_array_fixed_test.v
@@ -1,0 +1,7 @@
+fn test_main() {
+	mut space := [2][2][]int{}
+	space[1][1] << 123
+	space[0][0] << 321
+	assert space[1][1] == [123]
+	assert space[0][0] == [321]
+}

